sgl-router/src/tool_parser/parsers/minimax_m2.rs

```diff
@@ -0,0 +1,712 @@
+use std::{collections::HashMap, fmt::Write as FmtWrite, sync::LazyLock};
+
+use async_trait::async_trait;
+use regex::Regex;
+use serde_json::Value;
+
+use crate::{
+    protocols::common::Tool,
+    tool_parser::{
+        errors::{ParserError, ParserResult},
+        parsers::helpers,
+        traits::ToolParser,
+        types::{FunctionCall, StreamingParseResult, ToolCall, ToolCallItem},
+    },
+};
+
+// Static regex compilation - compiled once on first use
+static THINKING_REGEX: LazyLock<Regex> =
+    LazyLock::new(|| Regex::new(r"(?s)<think(?:ing)?>.*?</think(?:ing)?>").expect("Valid regex"));
+
+/// MiniMax M2 format parser for tool calls
+///
+/// Handles the MiniMax M2 specific format:
+/// `<minimax:tool_call><invoke name="func"><parameter name="key">value</parameter></invoke></minimax:tool_call>`
+///
+/// Features:
+/// - Namespaced XML tags (`minimax:tool_call`)
+/// - Function wrapped in `<invoke name="...">` tags
+/// - Parameters as `<parameter name="key">value</parameter>`
+/// - Incremental JSON streaming for parameters
```

Comment:
Can we have a link of the references to the call format? It is easier to validate and for developers to catch up

Reply:
nit: https://huggingface.co/MiniMaxAI/MiniMax-M2?chat_template=default